engines/stark/ui/menu/saveloadmenu.cpp

```diff
@@ -0,0 +1,276 @@
+
+#include "gui/message.h"
+
+namespace Stark {
+
+SaveLoadMenuScreen::SaveLoadMenuScreen(Gfx::Driver *gfx, Cursor *cursor, Screen::Name screenName) :
+		StaticLocationScreen(gfx, cursor, "LoadSaveLocation", screenName) {
+}
+
+SaveLoadMenuScreen::~SaveLoadMenuScreen() {
+}
+
+void SaveLoadMenuScreen::open() {
+	StaticLocationScreen::open();
+
+	_page = StarkSettings->getIntSetting(Settings::kSaveLoadPage);
+
+	_widgets.push_back(new StaticLocationWidget(
+			"loadsavebg",
+			nullptr,
+			nullptr));
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"back to index",
+			CLICK_HANDLER(SaveLoadMenuScreen, backHandler),
+			nullptr));
+	_widgets.back()->setupSounds(0, 1);
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"Cancel",
+			CLICK_HANDLER(SaveLoadMenuScreen, backHandler),
+			nullptr));
+	_widgets.back()->setupSounds(0, 1);
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"SaveText",
+			nullptr,
+			nullptr));
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"LoadText",
+			nullptr,
+			nullptr));
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"Back",
+			CLICK_HANDLER(SaveLoadMenuScreen, prevPageHandler),
+			nullptr));
+	_widgets.back()->setupSounds(0, 1);
+	_widgets.back()->setTextColor(_textColorBlack);
+	_widgets.back()->setVisible(_page > 0);
+	
+	_widgets.push_back(new StaticLocationWidget(
+			"Next",
+			CLICK_HANDLER(SaveLoadMenuScreen, nextPageHandler),
+			nullptr));
+	_widgets.back()->setupSounds(0, 1);
+	_widgets.back()->setTextColor(_textColorBlack);
+	_widgets.back()->setVisible(_page < 10);
+
+	loadSaveData(_page);
+}
+
+void SaveLoadMenuScreen::close() {
+	ConfMan.flushToDisk();
+	StaticLocationScreen::close();
+}
+
+void SaveLoadMenuScreen::backHandler() {
+	StarkUserInterface->backPrevScreen();
+}
+
+void SaveLoadMenuScreen::checkError(Common::Error error) {
+	if (error.getCode() != Common::kNoError) {
+		GUI::MessageDialog dialog(error.getDesc());
+		dialog.runModal();
+	}
+}
+
+void SaveLoadMenuScreen::removeSaveDataWidgets() {
+	assert(_widgets.size() == 16);
+
+	for (int i = 0; i < 9; ++i) {
+		delete _widgets.back();
+		_widgets.pop_back();
+	}
+}
+
+void SaveLoadMenuScreen::loadSaveData(int page) {
+	for (int i = 0; i < 9; ++i) {
+		_widgets.push_back(new SaveDataWidget(i + page * 9, _gfx, this));
+	}
+}
+
+void SaveLoadMenuScreen::changePage(int page) {
+	assert(page >= 0 && page <= 10);
+
+	removeSaveDataWidgets();
+	loadSaveData(page);
+
+	_widgets[kWidgetBack]->setVisible(page > 0);
+	_widgets[kWidgetNext]->setVisible(page < 10);
+
+	StarkSettings->setIntSetting(Settings::kSaveLoadPage, page);
+	_page = page;
+}
+
+void SaveMenuScreen::open() {
+	SaveLoadMenuScreen::open();
+	_widgets[kWidgetLoadText]->setVisible(false);
+}
+
+void SaveMenuScreen::onSlotSelected(int slot) {
+	int chapter = StarkGlobal->getCurrentChapter();
+
+	Common::String desc;
+	if (chapter == 0) {
+		desc = "Prologue";
+	} else {
+		desc = Common::String::format("Chapter %d", chapter);
```

Comment:
Ultimately, chapter names will have to come from `chapters.ini` so they are localized. But I guess having them hardcoded is fine for the first version. The original engine uses the string before the colon in `chapters.ini`.